### PR TITLE
Test groovylint with Groovy 4.x

### PR DESCRIPTION
**Pull request by nre-ableton**
- Preparations for Groovy 4 support
- Test groovylint with Groovy 4.x

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -17,10 +17,14 @@ devToolsProject.run(
...
-    String mirrorHost = 'groovy.jfrog.io/artifactory/dist-release-local/groovy-zips'
-    sh "curl -o ${groovy3Zip} https://${mirrorHost}/${groovy3Zip}"
-    unzip(zipFile: groovy3Zip)
+    data['groovy4Version'] = '4.0.2'
+
+    [data.groovy3Version, data.groovy4Version].each { groovyVersion ->
+      String groovyZip = "apache-groovy-binary-${groovyVersion}.zip"
+      String mirrorHost = 'groovy.jfrog.io/artifactory/dist-release-local/groovy-zips'
+      sh "curl -o ${groovyZip} https://${mirrorHost}/${groovyZip}"
+      unzip(zipFile: groovyZip)
```
Comment (mst-ableton) [inline Jenkinsfile:26]: I guess the groovy zips unzip to their own folders so we don't have to worry about collisions.

**Review by mst-ableton — APPROVED** [review]
lgtm, nice!

File: Jenkinsfile (hunk 1 above)
Comment (nre-ableton) [reply Jenkinsfile:26]: Yea, exactly. Note that we specify the groovy home path (which contains the version number) below when running the script.

**Comment by nre-ableton** [comment]
@ablbot rebase

**Pull request merged**